asadiqbal08/Updated the Receipt Page with additional Details

asadiqbal08 opened the pull request:
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
fixes: #531   
##### NOTE : Need to add some serializer tests.

#### What's this PR do?
As a user, I'd like to have enough information on the receipt so that I can share it with an employer for reimbursement.

#### How should this be manually tested?
- Visit the receipt page after an enrolment or by visiting `/orders/receipt/{RECEIPT_ID}/`
- Verify the required data is present as requested over the issue.


#### Screenshots (if appropriate)

![screencapture-localhost-8013-orders-receipt-2-2022-05-20-13_52_11](https://user-images.githubusercontent.com/7334669/169511833-c7d758e5-34b8-4b3b-b6b6-75e67e4418f4.png)

![screencapture-localhost-8013-orders-receipt-2-2022-05-20-15_41_51](https://user-images.githubusercontent.com/7334669/169511839-0ccdf129-85d5-44e0-9003-3e92e397130e.png)

pdpinch commented:
It seems like we're working on this in two places, and we should be sure the information (and maybe the styling?) is the same in both places. 

See #558

pdpinch commented:
A couple comments on the screenshot: 

- can we display the discount code?
- If the address is actually coming from the Cybersource receipt, and not the user model, it should be displayed under "payment information" instead of "customer information" 
- The font seems off to me, but I know we don't have a final design for this page. Please just make sure we're using the same basic body styles as the rest of the site.

asadiqbal08 commented:
I guess, Discount is there in [code](https://github.com/mitodl/mitxonline/pull/578/files#diff-17a03998399852a17aa3600e93122905838e5d4536942be8bf5c75349589c78eR163-R165) but hidden behind a condition

asadiqbal08 commented:
I updated the screenshots, You can see the: 
- `Discount code` and `Discount text` there. 
- I removed the conditional way for Discount. 
- I manually added values in model so one can ignore the actual factual values.

jkachel commented on frontend/public/scss/receipt.scss:
```diff
@@ -0,0 +1,257 @@
+  padding: 20px 15px;
+  margin: 0 auto;
+  max-width: 1018px;
+  font: 500 18px/23px 'Rajdhani', Times New Roman, serif;
```
Can this `font` rule (and where it appears elsewhere in this file - there's 11 total occurrences of Rajdhani in the file) be removed? This should use the standard fonts from the base layout.scss.

jkachel commented on frontend/public/scss/receipt.scss:
```diff
@@ -0,0 +1,257 @@
+  .footer,
+  .header-holder {
+    display: none !important;
+  }
```
Can you add "header" (the tag) to this list? It'd be good to not have the site header on the printed version of the receipt.

asadiqbal08 replied:
Removed. Done. 

<img width="1161" alt="Screenshot 2022-05-26 at 4 33 19 PM" src="https://user-images.githubusercontent.com/7334669/170479988-2edf8b0b-1b63-44ae-bea3-ae911cdf2de8.png">

jkachel reviewed (changes requested):
Few small layout changes - otherwise the functionality looks good to me.

jkachel commented on frontend/public/src/components/ReceiptPageDetailCard.js:
```diff
@@ -0,0 +1,225 @@
+                  <tr key={line.product.purchasable_object.readable_id}>
+                    <td>
+                      <div>
+                        {line.product.purchasable_object.title} <br />
```
Can you update this to get the course's readable_id rather than the one for the course run? Also, can you add the `course_number` to the beginning of the title (if it's a course that's been purchased)? That will make it match the email receipt code. (Take a look at the `TransactionLineSerializer` to see how we're doing it there.)

asadiqbal08 replied:
@jkachel If I am not wrong, It should not be CourseRun instead of a course here ?

asadiqbal08 replied:
@jkachel If I am not wrong, It should not be CourseRun instead of a course here ?

asadiqbal08 replied:
You mean that `line.product.purchasable_object.course.readable_id` but it can be a program.

asadiqbal08 replied:
@jkachel Do you mean the change set as in [commit here](https://github.com/mitodl/mitxonline/pull/578/commits/4e02dd6f4633213382ae113a4edc1bfde1611fbe) ?

asadiqbal08 replied:
<img width="1102" alt="Screenshot 2022-05-27 at 3 58 44 PM" src="https://user-images.githubusercontent.com/7334669/170686935-824266a0-5c97-43b2-bb0f-80f8909c8368.png">


Showing like this one.

jkachel replied:
Yup - tested with an order locally and looks good to me.

jkachel reviewed (changes requested):
One more change after reviewing against the email receipt.

jkachel reviewed (approved):
Flow types need fixing but this is good to go once that's done. LGTM 👍
